### PR TITLE
Rename trailing suffix to use ROS names

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
Renamed the folders to use ROS based names such as noetic-foxy or humble to create Docker images.

Currently named the focal into noetic-foxy to indicate both exist. 
In the future, in branch `humble-develop`, only `noetic` folder will exist.

I also rearranged humble Dockerfile to actually use caching (the metadata changes everytime when we use the build-image.sh script, so the image starts from the beginning since the metadata is declared in the beginning stages). 

## Related GitHub Issue
NA

## Related Jira Key
ARC-155 and ARC-154

## Motivation and Context
Upgrade to Humble

## How Has This Been Tested?
Ci builds fine without issues

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
-| Dockerhub Focal Build - Develop | Dockerhub Focal Build - Release | Dockerhub Jammy Build - Develop | Dockerhub Jammy Build - Release |
+| Dockerhub Noetic/Foxy Build - Develop | Dockerhub Noetic/Foxy Build - Release | Dockerhub Humble Build - Develop | Dockerhub Humble Build - Release |
```
Comment (adev4a) [inline README.md:1]: Can we keep this consistent to Noetic-Foxy everywhere?

File: docker/build-image.sh [hunk 2]
```diff
@@ -98,25 +98,25 @@ TAGS=()
...
 # TODO, distinguish with suffix when Humble is fully integrated
-# until then focal will have no suffix and be the main image
+# until then noetic-foxy will have no suffix and be the main image
 # https://usdot-carma.atlassian.net/browse/ARC-227
-if [ "$BUILD_FOCAL" = true ]; then
-    echo "Building carma-base focal image"
-    build_image "focal/Dockerfile" "" #replace with "-focal"
+if [ "$BUILD_NOETIC_FOXY" = true ]; then
+    echo "Building carma-base noetic-foxy image"
+    build_image "noetic-foxy/Dockerfile" "" #replace with just "-noetic"
```
Comment (adev4a) [inline docker/build-image.sh:114]: Is this replace comment supposed to be after the migration to humble is complete? I think its not a required comment, but if we are adding it, we should probably update it to something like "replace with just "-noetic" after ros2 is migrated to humble"

File: README.md (hunk 1 above)
Comment (MishkaMN) [reply README.md:1]: Replaced to Noetic-Foxy

File: docker/build-image.sh (hunk 2 above)
Comment (MishkaMN) [reply docker/build-image.sh]: Specified to your suggestion.

**Review by adev4a — APPROVED** [review]

**Pull request merged**